### ISSUE TITLE
Failed to download Intel Intrinsic Guide data

### ISSUE THREAD

**Issue by thorax58** [Sibras/ShiftIntrinsicGuide#1]
Hi.  I grabbed the precompiled binary package and unpacked it on my Windows system.  When I run the program I get the error shown in the title.  I know the data is accessible because I can reach it on the Intel site via web browser.

Not sure how to proceed.  Thanks.

Steve.

**Comment by Sibras** [comment]
I just tested this with a fresh download of the latest release and it worked fine. Perhaps there was an intermittent issue with the intel site when you first ran it and trying again may correct the issue

**Comment by Sibras** [comment]
Alternatively you can also just manually download the corresponding file and save it next to the executable and it will pick it up. The intel intrinsics come from [https://www.intel.com/content/dam/develop/public/us/en/include/intrinsics-guide/data-latest.xml](https://www.intel.com/content/dam/develop/public/us/en/include/intrinsics-guide/data-latest.xml) and should be saved as `intrin.xml`
The uops info comes from [https://www.uops.info/instructions.xml](https://www.uops.info/instructions.xml) and should be saved as `uops.xml`
This shouldnt be necessary but if theres an issue with your firewall blocking the app but not the browser this is one way to get around it (or just unblock it in your firewall)

**Comment by thorax58** [comment]
I tried it on another machine and it worked fine.  Thanks for the workaround though.

And finally, thanks for a great utility.  Wish I'd found it sooner. 🙂

Steve.
________________________________

**Comment by thorax58** [comment]
Hi Matthew,

The applications is great but I do have some suggestions for you to consider (in no specific order):


  1.
Would be nice to copy text from the intrinsic display.  For example if I search and expand the entry for __mm512_multishift_epi64_epi8 I'd like to be able to copy, for example, the perf information or the instruction name.  However, the click is used to expand/collapse the entry so can't be used for copying text.
  2.
Searching by instruction doesn't work for me.  If I enter "vperm"<CR> or "vpmulti"<CR>, for example, I get a blank screen.
  3.
I have a firewall issue so I have to download the intrinsic/uops data the manually as you showed earlier.  I have to do this every time I go into the application as it won't use the existing data if the connect to download fails.  Would be nice if it failed to connect it would just keep using the data it already has locally.
  4.
Some of the Description texts don't wrap and are displayed in a single line making the last half unreadable.  For example the Description text for _mm256_maskz_gf2p8affineinv_epi64_epi8 doesn't wrap.
  5.
The app losing focus causes the current open intrinsic to collapse.  Would be nice to have the intrinsic visible while working in another window.
  6.
Do the search as each character is typed rather than wait for <CR>.  (Similar to what Intel's Intrinsic Guide does.)  Would also be nice to ignore leading/trailing whitespace for the search.  (When I cut/paste from a Word spec into the app  always puts a trailing space which throws off the search.)


I've been using 1.0.5 but saw there was a 1.0.6 so I grabbed it and confirmed the behaviours are the same wrt my suggestions above.

Thanks again for a great app.

Steve
________________________________

**Comment by Sibras** [comment]
Ive added a fix for having to redownload the data cache every time. For the other items ill address them in #2
